daq_macros.py

```diff
@@ -2260,6 +2377,58 @@ def gotoMaxRaster(rasterResult,multiColThreshold=-1,**kwargs):
     z = hotCoords["z"]
     logger.info("goto " + str(x) + " " + str(y) + " " + str(z))
 
+    if "rasterRequest" in kwargs and autoRasterFlag:
+      # Update the raster request with the location of the max raster image and co-ordinates.
+      # NOTE: In the optimized autoraster, two 2D rasters are collected, face on and orthogonal.
+      # Both rasters have the same number of columns, we save the column number of the hot cell (max_col)
+      # in the face on raster then look for the hot cell in the orthogonal raster in the same 
+      # column. This is to emulate the original autoraster but leaves room for other ways of selecting
+      # points for standard collection
+      rasterDef = kwargs["rasterRequest"]["request_obj"]["rasterDef"]
+
+      col = get_raster_max_col(rasterDef, max_index)
+
+      
+      # Set the column index for the face on raster
+      if max_col is None:
+        max_col = col
+        face_on_max_coords = (x, y, z)
+      else: 
+        # max_col should be available for orthogonal rasters
+        # Find maximum in col defined in max_col and then reset max_col
+        indices = get_flattened_indices_of_max_col(rasterDef, max_col)
+        ceiling = 0
+        # Loop through all the cells that belong to the column corresponding to max_col
+        for index in indices:
+          try:
```

Comment:
A similar loop happens in this function with does the exact same thing but it handles multicol cases in there as well. I will refactor this function because it is a pain for me to add or modify it due to the number of loops and if conditions

Reply:
Fixed by refactoring function here: fdf7306cd9bde3739dc4dcdd103f6ed1fcdc1f8b